lib/index.js

```diff
@@ -59,21 +59,26 @@ const baseField = (_prop) => ({
 });
 
 
-const expandFields = (schema) => Object
+const expandFields = (schema, parentProp) => Object
   .keys(schema)
   .reduce((obj, prop) => {
     obj[prop] = !prop.startsWith('_')
-      ? toField(schema, prop)
+      ? toField(schema, prop, parentProp ? `${parentProp}.${prop}` : prop)
       : schema[prop];
     return obj;
   }, {});
 
 
-const toField = (schema, prop) => {
-  let field = {};
+const toField = (schema, prop, name) => {
+  let field = {
+    _name: name
+  };
```

Comment:
If you name the name arg `_name` you can do:

```
let field = {
   _name
};
```

Also lower down instead of using `field._name` you can just pass `_name` unless you're mutating it somewhere